libflux/flux-core/src/formatter/tests.rs

```diff
@@ -926,6 +926,10 @@ fn parens() {
     assert_format("(a() |> b()) ^ 3", "a() |> b() ^ 3");
     assert_format("1 ^ (a() |> b())", "1 ^ a() |> b()");
     assert_unchanged("(1 ^ a()) |> b()");
+    assert_format(
+        r#"qux = (r) => "foo: ${r._rab} is: " + (if r.bar then "bar" else "baz")"#,
+        r#"qux = (r) => "foo: ${r._rab} is: " + (if r.bar then "bar" else "baz")"#,
+    );
```

Comment:
This could be replaced with `assert_unchanged`, right? Using that would help a bit with readability.

This test is related to how you added precedence for `if/then/else`, right?

Since you've also switched to using `format_child_with_parens` in the formatter for the `test`, `consequent` and `alternate`, is there a test case you could add that shows how the behavior changed as a result?

Reply:
I have updated it with assert_unchanged.